fed/_private/global_context.py

```diff
@@ -28,27 +35,40 @@ def next_seq_id(self) -> int:
     def get_cleanup_manager(self) -> CleanupManager:
         return self._cleanup_manager
 
-    def job_name(self) -> str:
+    def get_job_name(self) -> str:
         return self._job_name
 
+    def get_failure_handler(self) -> Callable[[], None]:
+        return self._failure_handler
+
+    def acquire_shutdown_flag(self) -> bool:
+        with threading.Lock():
+            if self._atomic_shutdown_flag:
+                self._atomic_shutdown_flag = False
+                return True
+            return False
+
 
 _global_context = None
 
 
-def init_global_context(job_name: str) -> None:
+def init_global_context(current_party: str,
+                        job_name: str,
+                        failure_handler: Callable[[], None] = None) -> None:
     global _global_context
     if _global_context is None:
-        _global_context = GlobalContext(job_name)
+        _global_context = GlobalContext(job_name, current_party, failure_handler)
 
 
 def get_global_context():
     global _global_context
-    if _global_context is None:
-        _global_context = GlobalContext()
+    # if _global_context is None:
```

Comment:
Why comment these lines out. BTW, if we should comment them, let's remove them.

Reply:
I don't think a `getter` should do initialization works, only `init_global_context` should do that.

Besides, when `fed.get` call crushes because of an error, other fed calls afterward will try to `get_global_context` in many places, and init the context again.